includes/class-post-type.php

```diff
@@ -0,0 +1,220 @@
+			"view_item"             => __( "View $this->singular", $this->text_domain ),
+			"search_items"          => __( "Search $this->plural", $this->text_domain ),
+			"not_found"             => __( "Not found", $this->text_domain ),
+			"not_found_in_trash"    => __( "Not found in Trash", $this->text_domain ),
+			"featured_image"        => __( "Featured Image", $this->text_domain ),
+			"set_featured_image"    => __( "Set featured image", $this->text_domain ),
+			"remove_featured_image" => __( "Remove featured image", $this->text_domain ),
+			"use_featured_image"    => __( "Use as featured image", $this->text_domain ),
+			"insert_into_item"      => __( "Insert into $this->singular_lower", $this->text_domain ),
+			"uploaded_to_this_item" => __( "Uploaded to this $this->singular_lower", $this->text_domain ),
+			"items_list"            => __( "$this->plural list", $this->text_domain ),
+			"items_list_navigation" => __( "$this->plural list navigation", $this->text_domain ),
+			"filter_items_list"     => __( "Filter $this->plural_lower list", $this->text_domain ),
+		);
+
+		$retval = apply_filters( 'ucf_research_labels', $retval );
+
+		return $retval;
+	}
+
+	/**
+	 * Returns the args array used in
+	 * registering the custom post type
+	 * @author Jim Barnes
+	 * @since 1.0.0
+	 * @return array The argument array
+	 */
+	private function args() {
+		$taxonomies = $this->taxonomies();
+
+		$args = array(
+			'label'               => __( $this->singular, $this->text_domain ),
+			'description'         => __( $this->plural, $this->text_domain ),
+			'labels'              => $this->labels(),
+			'supports'            => array( 'title', 'editor', 'excerpt', 'revisions' ),
+			'taxonomies'          => $taxonomies,
+			'hierarchical'        => false,
+			'public'              => true,
+			'show_ui'             => true,
+			'show_in_menu'        => true,
+			'show_in_rest'        => true,
+			'rest_base'           => 'research',
+			'menu_position'       => 8,
+			'menu_icon'           => 'dashicons-welcome-write-blog',
+			'show_in_admin_bar'   => true,
+			'show_in_nav_menus'   => true,
+			'can_export'          => true,
+			'has_archive'         => false,
+			'exclude_from_search' => false,
+			'publicly_queryable'  => true,
+			'capability_type'     => 'post'
+		);
+
+		$args = apply_filters( 'ucf_research_args', $args );
+
+		return $args;
+	}
+
+	/**
+	 * Function that returns the taxonomies
+	 * used by the custom post type
+	 * @author Jim Barnes
+	 * @since 1.0.0
+	 * @return array The array of taxonomies
+	 */
+	public function taxonomies() {
+		return apply_filters(
+			'ucf_research_taxonomies',
+			array()
+		);
+	}
+
+	/**
+	 * Appends the post meta to the `meta`
+	 * field on each post.
+	 * @author Jim Barnes
+	 * @since 1.0.0
+	 * @param WP_Post[] $posts The post array
+	 * @param WP_Query $query The query object. Added for convenience
+	 * @return WP_Post[] The post array
+	 */
+	public function meta( $posts, $query ) {
```

Comment:
Do we still need to be appending meta data to post objects manually?  IIRC post meta should be accessible as post object properties already: https://developer.wordpress.org/reference/functions/get_post_meta/#comment-1894

Reply:
Removed.